app/models/user.rb

```diff
@@ -1,2 +1,13 @@
 class User < ApplicationRecord
+  before_save :downcase_email
+
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: Settings.user.email.max_length}, format: {with: VALID_EMAIL_REGEX}
+
+  def downcase_email
+    self.email.downcase!
+  end
```

Comment:
recommend private this function
```suggestion
  has_secure_password
  
  private
  
  def downcase_email
    self.email.downcase!
  end
```